drivers/wifi/eswifi/eswifi_offload.c

```diff
@@ -20,62 +20,36 @@ LOG_MODULE_REGISTER(wifi_eswifi_offload);
 
 #include "eswifi.h"
 
-static int __select_socket(struct eswifi_dev *eswifi, u8_t idx)
+static inline int __select_socket(struct eswifi_dev *eswifi, u8_t idx)
 {
-	int err;
-
 	snprintf(eswifi->buf, sizeof(eswifi->buf), "P0=%d\r", idx);
-
-	err = eswifi_request(eswifi, eswifi->buf, strlen(eswifi->buf),
-			     eswifi->buf, sizeof(eswifi->buf));
-	if (err || !eswifi_is_buf_at_ok(eswifi->buf)) {
-		return -EIO;
-	}
-
-	return 0;
+	return eswifi_at_cmd(eswifi, eswifi->buf);
 }
 
-static int __read_data(struct eswifi_dev *eswifi, size_t len)
+static int __read_data(struct eswifi_dev *eswifi, size_t len, char **data)
 {
 	char cmd[] = "R0\r";
```

Comment:
I see that this older code, but should these (and similar in other funcs) be const?